clients/src/main/java/org/apache/kafka/clients/consumer/AcknowledgementCommitCallback.java

```diff
@@ -42,12 +44,16 @@ public interface AcknowledgementCommitCallback {
      *
      * @param exception The exception thrown during processing of the request, or null if the acknowledgement completed successfully.
      * <p><ul>
+     * <li> {@link AuthorizationException} if not authorized to the topic or group
      * <li> {@link InvalidRecordStateException} if the record state is invalid
-     * <li> {@link AuthorizationException} if not authorized to the topic of group
+     * <li> {@link NotLeaderOrFollowerException} if the leader had changed by the time the acknowledgements were sent
+     * <li> {@link DisconnectException} if the broker disconnected before the request could be completed
      * <li> {@link WakeupException} if {@link KafkaShareConsumer#wakeup()} is called before or while this function is called
      * <li> {@link InterruptException} if the calling thread is interrupted before or while this function is called
      * <li> {@link KafkaException} for any other unrecoverable errors
      * </ul>
+     * <p>Note that if the exception is a retriable exception, the acknowledgement could not be completed and the
```

Comment:
nit: For non-retribale exceptions it will fetch anyways again so I expect we mean that `even` for retriable exception, the fetch will happen again?
```suggestion
     * <p>Note that if the exception is even a retriable exception, the acknowledgement could not be completed and the
```

Reply:
I will reword.